webview-ui/src/ConnectAcrToCluster/ConnectAcrToCluster.tsx

```diff
@@ -0,0 +1,280 @@
+import { useEffect } from "react";
+import {
+    Acr,
+    Cluster,
+    InitialState,
+    Subscription,
+} from "../../../src/webview-contract/webviewDefinitions/connectAcrToCluster";
+import { Lazy, isLoaded, map as lazyMap } from "../utilities/lazy";
+import { useStateManagement } from "../utilities/state";
+import styles from "./ConnectAcrToCluster.module.css";
+import {
+    EventHandlerFunc,
+    ensureAcrRoleAssignmentLoaded,
+    ensureAcrsLoaded,
+    ensureClustersLoaded,
+    ensureSubscriptionsLoaded,
+} from "./state/dataLoading";
+import { ConnectAcrToClusterState, stateUpdater, vscode } from "./state/state";
+import { distinct } from "../utilities/array";
+import { ResourceSelector } from "../components/ResourceSelector";
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconDefinition, faCheckCircle, faClock, faSave, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { AcrRoleState } from "./state/stateTypes";
+
+export function ConnectAcrToCluster(initialState: InitialState) {
+    const { state, eventHandlers } = useStateManagement(stateUpdater, initialState, vscode);
+
+    const updates: EventHandlerFunc[] = [];
```

Comment:
While names like updates, lazeSubscriptions are not bad, they could be more descriptive. Consider using names that clearly describe what the variable is or does.

Reply:
Happy to discuss suggestions, otherwise I'll go with "not bad".

`lazySubscriptions` is a collection of subscriptions that are lazily-loaded. I.e. at any given point in time they might be loaded, or they might not. A more descriptive term might be `lazilyLoadedSubscriptions`. But that's a lot more typing and also inconsistent (since the type is called `Lazy` and I'd prefer not to rename the type and all usages of it).

Personally, I think `lazy` is adequate, and unlikely to cause confusion if you have some familiarity with the codebase.